Fix : DIG-177 FCM 웹 푸시 버그 수정

YEJINGO opened the pull request:
- 알림을 차단하고, 허용했을 때 새로운 FCM토큰이 발급되는데 이때, 전 토큰값이 그대로 DB에 저장되어 FCM 웹푸시가 되지 않음
    - 새로운 토큰이 들어왔을 때, 전 토큰값을 삭제하고 새로운 토큰을 저장

리팩토링
- FcmToken, FCMAlarm 서비스 분리
    - 토큰 생성/수정/삭제 로직과 웹푸시 알람 보내는 메서드 분리 
- FcmToken Entity 연관관계 수정 
    - @OneToOne -> @ManyToOne로 수정하여 User를 FK로 설정
    - 고유의 id값을 PK로 설정 
- ScheduledService 내용 수정
     - 운동일지가 비었을 때, 비어있다는 알림 추가

dunowljj commented on src/main/java/com/ogjg/daitgym/alarm/service/ScheduledService.java:
```diff
@@ -27,31 +27,43 @@ public class ScheduledService {
         for (FcmToken fcmToken : fcmTokens) {
             User user = fcmToken.getUser();
-            ExerciseJournal exerciseJournal = exerciseJournalRepository.findByUserAndJournalDate(user, LocalDate.now()).orElseThrow(NotFoundExerciseJournal::new);
+            Optional<ExerciseJournal> optionalExerciseJournal = exerciseJournalRepository.findByUserAndJournalDate(user, LocalDate.now());
+            ExerciseJournal exerciseJournal = optionalExerciseJournal.get();
```
exerciseJournal을 미리 get으로 꺼내는 이유가 있을까요?
꺼낸 직후 if문 분기에서 값이 존재하지 않을 경우에 대해 처리하고 있는데, else문에 도달하기 전에 예외가 발생하지 않을까 생각이 들었습니다!

YEJINGO replied:
수정했습니다!! 알려주셔서 감사합니다!

dunowljj commented on src/main/java/com/ogjg/daitgym/alarm/service/FcmTokenService.java:
```diff
@@ -0,0 +1,75 @@
+    private final FcmTokenRepository fcmTokenRepository;
+
+
+    public boolean getNotification(OAuth2JwtUserDetails oAuth2JwtUserDetails) {
+
+        User user = userHelper.findUserByEmail(oAuth2JwtUserDetails.getEmail());
+        log.info("USER : " + user);
+        fcmTokenRepository.findByUser(user).isPresent();
+        return fcmTokenRepository.findByUser(user).isPresent();
+
+    }
```
1. 30,31은 같은 내용인데 실수로 넣으신 것 같아요!
2. 이메일로 유저를 조회하고, 그 유저로 다시 fcm토큰을 조회하고있습니다. fcm 토큰이 email을 fk로 가지고 있으니, 바로 email로 조회하면 안되는지 궁금합니다.

YEJINGO replied:
1. 알려주셔서 감사합니다!
2. 바로 조회하는 로직으로 수정했습니다 감사합니다!

YEJINGO committed:
Fix : DIG-177 FCM 웹 푸시 버그 수정

- 알림을 차단하고, 허용했을 때 새로운 FCM토큰이 발급되는데 이때, 전 토큰값이 그대로 DB에 저장되어 FCM 웹푸시가 되지 않음
    - 새로운 토큰이 들어왔을 때, 전 토큰값을 삭제하고 새로운 토큰을 저장

리팩토링
- FcmToken, FCMAlarm 서비스 분리
    - 토큰 생성/수정/삭제 로직과 웹푸시 알람 보내는 메서드 분리
- FcmToken Entity 연관관계 수정
    - @OnetoOne -> @manytoone로 수정하여 User를 FK로 설정
    - 고유의 id값을 PK로 설정
- ScheduledService 내용 수정
    - 운동일지가 비었을 때, 비어있다는 알림 추가
- NotificationApiController 삭제, FcmTokenController 로 변경

dunowljj reviewed (approved):
고생하셨습니다 :)
